repose-aggregator/components/filters/herp/src/test/scala/org.openrepose.filters.herp/HerpFilterTest.scala

```diff
@@ -0,0 +1,335 @@
+    }
+    it("should log the configured service code") {
+      // given:
+      herpConfig.setServiceCode("some-service")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"ServiceCode\" : \"some-service\"")
+    }
+    it("should log the configured region") {
+      // given:
+      herpConfig.setRegion("some-region")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"Region\" : \"some-region\"")
+    }
+    it("should log the configured data center") {
+      // given:
+      herpConfig.setDataCenter("some-data-center")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"DataCenter\" : \"some-data-center\"")
+    }
+    it("should extract and log the request method") {
+      // given:
+      servletRequest.setMethod("POST")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"Method\" : \"POST\"")
+    }
+    it("should extract and log the request url") {
+      // given:
+      servletRequest.setAttribute("http://openrepose.org/requestUrl", "http://foo.com")
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"URL\" : \"http://foo.com\"")
+    }
+    it("should extract and log the request parameters") {
+      // given:
+      servletRequest.setAttribute("http://openrepose.org/queryParams", Map("foo" -> Array("bar", "baz")).asJava)
+
+      // when:
+      herpFilter.configurationUpdated(herpConfig)
+      herpFilter.doFilter(servletRequest, servletResponse, filterChain)
+
+      // then:
+      def logEvents = listAppender.getEvents
+      logEvents.size shouldBe 1
+      logEvents.get(0).getMessage.getFormattedMessage should include("\"Parameters\" : {  \"foo\" : [  \"bar\",  \"baz\",  ],  }")
```

Comment:
Ugh, the other solution is to wrap anything like this in an object that does a to string with the correct values, but that limits us, because then we are enforcing json syntax into it.